lib/src/components/joystick/joystick_action.dart

```diff
@@ -144,45 +159,47 @@ class JoystickAction {
         _rectAction = _rectAction.shift(diff);
       }
 
-      final double _intensity = dist / _tileSize;
+      final _intensity = dist / _tileSize;
 
-      _joystickController.joystickAction(
+      joystickController.joystickAction(
         JoystickActionEvent(
           id: actionId,
           event: ActionEvent.MOVE,
           intensity: _intensity,
-          radAngle: _radAngle,
+          radAngle: radAngle,
         ),
       );
     } else {
       if (_rectAction != null) {
-        final Offset diff = _dragPosition - _rectAction.center;
-        _rectAction = _rectAction.shift(diff);
+        final diff = _dragPosition - _rectAction.center.toVector2();
+        _rectAction = _rectAction.shift(diff.toOffset());
       }
     }
   }
 
-  void onReceiveDrag(DragEvent event) {
-    if (_dragging || !(_rectAction?.contains(event.initialPosition) ?? false)) {
-      return;
+  @override
+  bool containsPoint(Vector2 point) {
```

Comment:
these guys should really all be position components. but that is too much for this PR